internal/tablespace_spec.go

```diff
@@ -0,0 +1,147 @@
+package internal
+
+import (
+	"fmt"
+	"github.com/tinsane/tracelog"
+	"github.com/wal-g/wal-g/utility"
+	"path/filepath"
+)
+
+const (
+	BasePrefix  = "base_prefix"
+	Tablespaces = "tablespaces"
+)
+
+// The mandatory keys for this map are "base_prefix" and "tablespaces".
+// "base_prefix" contains Location of pg_data folder.
+// "tablespaces" contains array of keys, which also happen to be names of tablespace folders.
+// The rest keys should be these names of tablespace folders and values should be TablespaceLocation structs.
+type TablespaceSpec map[string]interface{}
```

Comment:
I get thought about unmarshaling. So maybe it would be better to separate this logic. I mean that we can convert struct to map before marshaling and convert the map to struct after unmarshaling

Reply:
I decided to use the struct and implement interface so that this struct could be marshalled and unmarshalled.